irohad/ametsuchi/impl/postgres_wsv_command.cpp

```diff
@@ -17,335 +17,379 @@
 
 #include "ametsuchi/impl/postgres_wsv_command.hpp"
 
+#include <exception>
 #include <iterator>
 #include <numeric>
 
 namespace iroha {
   namespace ametsuchi {
-
     PostgresWsvCommand::PostgresWsvCommand(pqxx::nontransaction &transaction)
         : transaction_(transaction), log_(logger::log("PostgresWsvCommand")) {}
 
-    bool PostgresWsvCommand::insertRole(const std::string &role_name) {
+    const char *kUnsuccesfulAlloc = "Unable to alloc: {}";
+
+    template <typename T>
+    std::string try_quote(PostgresWsvCommand *that,
+                          std::initializer_list<T> l) noexcept {
       try {
-        transaction_.exec("INSERT INTO role(role_id) VALUES ("
-                          + transaction_.quote(role_name)
-                          + ");");
-      } catch (const std::exception &e) {
-        log_->error(e.what());
-        return false;
+        return try_quote(
+            that,
+            std::accumulate(l.begin(),
+                            l.end(),
+                            std::string(),
+                            [](std::string s, auto next) { return s + next; }));
+      } catch (std::bad_alloc &e) {
+        that->log_->error(kUnsuccesfulAlloc, e.what());
+        std::terminate();
       }
-      return true;
     }
 
-    bool PostgresWsvCommand::insertAccountRole(const std::string &account_id,
-                                               const std::string &role_name) {
+    template <typename T>
+    std::string try_quote(PostgresWsvCommand *that, const T &s) noexcept {
       try {
-        transaction_.exec(
-            "INSERT INTO account_has_roles(account_id, role_id) VALUES ("
-            + transaction_.quote(account_id)
-            + ", "
-            + transaction_.quote(role_name)
-            + ");");
-      } catch (const std::exception &e) {
-        log_->error(e.what());
-        return false;
+        return that->transaction_.quote(s);
+      } catch (std::bad_alloc &e) {
+        that->log_->error(kUnsuccesfulAlloc, e.what());
+        std::terminate();
       }
-      return true;
     }
 
-    bool PostgresWsvCommand::deleteAccountRole(const std::string &account_id,
-                                               const std::string &role_name) {
+    std::string try_concat(PostgresWsvCommand *that,
+                           std::initializer_list<const char *> l) noexcept {
```

Comment:
are you sure that T=const char* ? maybe std::string is better?

Reply:
Nope, std::string may throw somewhere